### PR TITLE
fix faster_rcnn sample crashed at PoolingInvoker on Windows7(x64).

### PR DESCRIPTION
**Pull request by bygreencn**
fix faster_rcnn sample crashed at PoolingInvoker operator() of pooling_layer on Windows7(x64).
Change  ofsbuf from std::vector<int> to AutoBuffer<int>, since at some pooling_layer ofsbuf is not initialized and "const int* ofsptr = &ofsbuf[0];" will crash.

### CODE REVIEW

**Comment by vpisarev** [comment]
@bygreencn, thanks for the patch! let's fix in a slightly different way, see below

File: modules/dnn/src/layers/pooling_layer.cpp [hunk 1]
```diff
@@ -287,7 +287,7 @@ class PoolingLayerImpl : public PoolingLayer
             bool compMaxIdx = computeMaxIdx;
 
 #if CV_SIMD128
-            const int* ofsptr = &ofsbuf[0];
+            const int* ofsptr = (const int*)ofsbuf;
```
Comment (vpisarev) [inline modules/dnn/src/layers/pooling_layer.cpp]: `const int* ofsptr = ofsbuf.empty() ? 0 : &ofsbuf[0];` is the better solution

File: modules/dnn/src/layers/pooling_layer.cpp [hunk 2]
```diff
@@ -230,7 +230,7 @@ class PoolingLayerImpl : public PoolingLayer
         Size kernel, stride, pad;
         int nstripes;
         bool computeMaxIdx;
-        std::vector<int> ofsbuf;
+        AutoBuffer<int> ofsbuf;
```
Comment (vpisarev) [inline modules/dnn/src/layers/pooling_layer.cpp]: let's keep `std::vector<>`

Comment (alalek) [reply modules/dnn/src/layers/pooling_layer.cpp]: If we need some alignment requirement, we could use `AutoBuffer<int, 0>` to use OpenCV `fastAlloc()`.

File: modules/dnn/src/layers/pooling_layer.cpp (hunk 1 above)
Comment (bygreencn) [reply modules/dnn/src/layers/pooling_layer.cpp]: Perfect, I think it is OK.

File: modules/dnn/src/layers/pooling_layer.cpp (hunk 2 above)
Comment (bygreencn) [reply modules/dnn/src/layers/pooling_layer.cpp]: I think it is OK.

**Comment by vpisarev** [comment]
@alalek, there is no any alignment requirements of ofsbuf/ofsptr. @bygreencn, please, change AutoBuffer back to std::vector and retrieve the data pointer in the way I suggested.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**